### PR TITLE
manifest: update default lto for release and bench to "thin"

### PR DESCRIPTION
**Pull request by matthiaskrgr**
[redacted]

### CODE REVIEW

File: src/doc/src/reference/manifest.md [hunk 1]
```diff
@@ -288,6 +288,8 @@ along with the defaults for each profile.
 [profile.dev]
 opt-level = 0      # controls the `--opt-level` the compiler builds with.
                    # 0-1 is good for debugging. 2 is well-optimized. Max is 3.
+                   # 's' will only run optimizations that do not increase
+                   # binary size, 'z' will optimize for size aggressively.
```
Comment (matklad) [inline src/doc/src/reference/manifest.md]: Aren't `-s` and `-z` nightly only?

Comment (matthiaskrgr) [reply src/doc/src/reference/manifest.md]: Hm, I never used any ````-Z ..```` flag to enable these, but lemme check...

Comment (matthiaskrgr) [reply src/doc/src/reference/manifest.md]: You're right!

Comment (matthiaskrgr) [reply src/doc/src/reference/manifest.md]: Updated, thanks!

**Comment by matklad** [comment]
@bors r+

Thanks!

**Comment by alexcrichton** [comment]
Oh unfortunately this isn't actually correct, specifying `lto = 'thin'` is not the same as specifying `lto = false`

**Comment by matthiaskrgr** [comment]
Oh, I thought that bench and release use thinlto by default and was trying to reflect that in the manifest.

**Comment by alexcrichton** [comment]
There are multiple flavors of LTO/ThinLTO, but `lto = 'thin'` is not what happens by default.

**Comment by matthiaskrgr** [comment]
Ok, my bad, please revert then.

**Comment by matthiaskrgr** [comment]
Reverted in #5225

**Pull request merged**